### DISCUSSION TITLE
Invalid Positioin in decode for YieldThisYear2 (Vaillant VWL 125/5)

### DISCUSSION THREAD

**Discussion by HSoft99** [john30/ebusd#1417] [Q&A]
Hello,

my configuration runs successful and I get a lot of values. But there are no values for yearly information.

What's wrong? In my log-file there are a lot of elements like this "[update notice] received unknown MS cmd: ..."

Have anybody some information to solve my problem?

**For example**
ebusctl read YieldThisYear2
   ERR: invalid position in decode

ebusctl find -f YieldThisYear2
  r,hmu,YieldThisYear2,,,08,b516,1a010200,,s,IGN:1,,,,month,s,D1B,,kWh,,,s,IGN:1,,,,energy,s,UIN,,kWh,

ebusctl find
  hmu YieldThisYear2 =  (ERR: invalid position for 3108b516041a010200 / 00)

**Confguration:**

_ebusctl scan result_
08;Vaillant;HMU00;0902;5103
15;Vaillant;CTLV3;0709;3704
26;Vaillant;VR_71;0201;0503
76;Vaillant;VWZIO;0902;5103
f6;Vaillant;NETX3;0124;0404

_ebusctl info_
version: ebusd 24.1.24.1
update check: OK, device firmware v5-c6 1[4a1b] up to date, vaillant/26.vr_71.csv: different version available
device: 192.168.177.120:9999, TCP, enhanced, firmware 1.1[4a1b].1[4a1b]
signal: acquired
symbol rate: 42
max symbol rate: 209
min arbitration micros: 1
max arbitration micros: 32
min symbol latency: 4
max symbol latency: 19
scan: finished
reconnects: 0
masters: 5
messages: 524
conditional: 0
poll: 0
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0902;HW=5103", loaded "vaillant/08.hmu.HW5103.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV3;SW=0709;HW=3704", loaded "vaillant/15.ctlv3.csv"
address 26: slave, scanned "MF=Vaillant;ID=VR_71;SW=0201;HW=0503", loaded "vaillant/26.vr_71.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0902;HW=5103", loaded "vaillant/76.vwzio.csv"
address f1: master #10
address f6: slave #10, scanned "MF=Vaillant;ID=NETX3;SW=0124;HW=0404", loaded "vaillant/f6.netx3.csv"

**Comment by chrizzzp** [comment]
> ebusctl find -f YieldThisYear2 r,hmu,YieldThisYear2,,,08,b516,1a010200,,s,IGN:1,,,,month,s,D1B,,kWh,,,s,IGN:1,,,,energy,s,UIN,,kWh,
> 
> ebusctl find hmu YieldThisYear2 = (ERR: invalid position for 3108b516041a010200 / 00)

Same for the VWL 105/5. I think the definitions with the default `*r,,,,,,B516,1A,,,IGN:1,,,,,,,,,,,,,,,,,,,,,` don't work for these units (if they exist, correct registers still have to be identified).  

For me the following do work:

```
*r,,,,,,B51A,05,,,IGN:3,,,,,,,,,,,,,,,,,,,,,
*w,,,,,,B51A,06,,,,,,,,,,,,,,,,,,,,,,,,
#?? *w,,,,,,B51A,05,,,,,,,,,,,,,,,,,,,,,,,,
r,,YieldHcDay,,,,,FF3200,,,energy,10,kWh,,,,,,,,,,,,,,,,,,,
r,,YieldCoolingDay,,,,,FF3201,,,energy,10,kWh,,,,,,,,,,,,,,,,,,,
r,,YieldHwcDay,,,,,FF3202,,,energy,10,kWh,,,,,,,,,,,,,,,,,,,
r,,YieldHcMonth,,,,,FF320E,,,energy,10,kWh,,,,,,,,,,,,,,,,,,,
r,,CopHcMonth,,,,,FF320F,,,UIN,10, ,,,,,,,,,,,,,,,,,,,
r,,YieldHc,,,,,FF3210,,,energy,,kWh,Total Yield Heating Circuit,,,,,,,,,,,,,,,,,,
r,,CopHc,,,,,FF3211,,,UIN,10, ,Total Cop Heating Circuit,,,,,,,,,,,,,,,,,,
r,,YieldHwcMonth,,,,,FF3212,,,energy,10,kWh,,,,,,,,,,,,,,,,,,,
r,,CopHwcMonth,,,,,FF3213,,,UIN,10, ,,,,,,,,,,,,,,,,,,,
r,,YieldHwc,,,,,FF3216,,,energy,,kWh,Total Yield Hot Water Circuit,,,,,,,,,,,,,,,,,,
r,,CopHwc,,,,,FF3217,,,UIN,10,,Total Cop Hot Water Circuit,,,,,,,,,,,,,,,,,,
```

**Reply by HSoft99** [reply]
Thanks @chrizzzp for the information, now works the next block.  Do you get the information automatically? I get the information  only by manual read per ebusctl read ...

My issue is a different block in my config file who is not working

# Ecologic energy Yields,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
#--------------------------------------------------
*r,,,,,,"B516","1A01",,,IGN:1,,,,,,,,,,,,,,,,,,,,,
#--------------------------------------------------
r,,YieldThisYear1,,,,,0100,month,,D1B,,,,,,IGN:1,,,,,,energy,,,,,,,,, until ...Year12

r,,EnergyConsumptionThisYear1,,,,,020100,month,,D1B,,kWh,,,,IGN:1,,,,,,energy,,,,,,,,, until ...Year 12


# Electric power consumption,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
#--------------------------------------------------
*r,,,,,,"B516","1A02",,,IGN:1,,,,,,,,,,,,,,,,,,,,,
#--------------------------------------------------
r,,ConsumptionThisYear1,,,,,0100,month,,D1B,,,,,,IGN:1,,,,,,energy,,,,,,,,, until ...Year12

**Reply by UnderPalms** [reply]
I have the same problem with another device (VPM20). There, "ec.sol.csv" gets loaded and it seems that the registers to not fit at all.
however, I recently tried to identify the registers by myself. Wrote a script which calls the command "ebusctl hex [sender address, for me 31] ec509b02[2 Bytes trying all registers from 00 00 to ff ff]. 
Maybe you can do the same. I reached to a stage where I identified a lot of values changing over time  on register 07. However, I am not clear on converting the bytes (5 bytes) into decimal figures.

**Reply by chrizzzp** [reply]
@HSoft99 

> Do you get the information automatically? I get the information only by manual read per ebusctl read ...

According to my logfiles they are only polled by the ebus adapter. Some of them I manually poll more often than others via Home Assistant (mqtt).

>My issue is a different block in my config file who is not working
>Ecologic energy Yields,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
>
>*r,,,,,,"B516","1A01",,,IGN:1,,,,,,,,,,,,,,,,,,,,,

As I wrote the message definitions targeting the 'B516,1A*' block don't work for our units.

**Reply by chrizzzp** [reply]
@UnderPalms 
> Wrote a script which calls the command "ebusctl hex [sender address, for me 31] ec509b02[2 Bytes trying all registers from 00 00 to ff ff].
> Maybe you can do the same. I reached to a stage where I identified a lot of values changing over time on register 07. However, I am > not clear on converting the bytes (5 bytes) into decimal figures.

Yes, this is definitely a way to go, but one should be careful as sending hex commands could actually issue write commands and thereby could change the configuration of the device or induce some unexpected behavior. 

To decode the response you need to figure out the data type. Tracking changes (and maybe visualizing them) is a good approach to correlate them to a certain device activity.

Could you post an example?

**Reply by HSoft99** [reply]
I start ebusd with the parameters --checkconfig --dumpconfig=csv –dumpconfigto=config.csv. This command returned a file with the configuration (config.csv). I hope that helps. I tried the configuration, but the result was the same (ERR: invalid position in decode). I don't understand why. @john30  is there something to note?

For example the item from config.csv
r,hmu,ConsumptionThisYear1,default *r,,08,b516,1a020100,value,s,IGN:1,,,,month,s,SCH,,,,ign,s,IGN:1,,,,energy,s,UIN,,kWh,

**Reply by chrizzzp** [reply]
I think the message definition does not match the response obtained. Could you post the result of

`
ebusctl f ConsumptionThisYear1`

**Reply by HSoft99** [reply]
Here the reuslts

ebusctl read ConsumptionThisYear1
ERR: invalid position in decode

ebusctl f ConsumptionThisYear1
hmu ConsumptionThisYear1 = (ERR: invalid position for 3108b516041a020100 / 00)

**Reply by chrizzzp** [reply]
> ERR: invalid position for 3108b516041a020100 / 00

Obviously, the message is not understood by the hmu (response 00), indicating the field address (b516 1a 020100) is not valid for this unit.

**Reply by HSoft99** [reply]
Thanks, good to know. But I'm suprised that I get the note with parameters --checkconfig --dumpconfig=csv –dumpconfigto=config.csv. I connect ebusd via [https://adapter.ebusd.eu/v5-c6/index.en](url). Is there a possibilty to get the available field addresses?

**Reply by chrizzzp** [reply]
> Thanks, good to know. But I'm suprised that I get the note with parameters --checkconfig --dumpconfig=csv –dumpconfigto=config.csv.

I'm not sure if I fully understand what you mean by the "note", but the configuration itself is probably OK (CSV), just the address for `ConsumptionThisYear1` is not. However, this address seems to be correct for other units, that's why it's still in the hmu.csv. There are more cases where the configuration in the CSVs works for some family of devices, while for other they don't. I think this has to be checked and confirmed for every kind of unit (by trial and error).

**Reply by HSoft99** [reply]
That's was my mistake. I thougt the result is for my configuration

**Reply by HSoft99** [reply]
> > Do you get the information automatically? I get the information only by manual read per ebusctl read ...
> 
> According to my logfiles they are only polled by the ebus adapter. Some of them I manually poll more often than others via Home Assistant (mqtt).

Hi @chrizzzp, 
I only used mqtt for reading and writing items. Can you tell me how to perform reading from ebusd using MQTT. Just for information, I use openhab as home system

**Reply by chrizzzp** [reply]
@HSoft99 
I meant the values are polled via ebus from the ebus-interface hardware (master address 31) and not by any other ebus device (in which case the ebus adapter would only be passively listening). If mqtt is enabled, ebusd then automatically passes the response to mqtt.

**Comment by fabian1512** [comment]
@HSoft99 : where did you find the file vaillant/f6.netx3.csv?
could you please upload it here? My config complains, the file for f6 is missing

thank you in advance

**Reply by HSoft99** [reply]
@fabian1512 here is the file, I hope it works in your configuration
[f6.netx3.csv](https://github.com/user-attachments/files/20354306/f6.netx3.csv)

**Reply by fabian1512** [reply]
Hi @HSoft99, 
many thanks for this, unfortunately the file did not produce the desired results.
Where did you get the file from? I have searched the entire Internet and have not found a source

**Reply by HSoft99** [reply]
Hi @fabian1512 ,
try and error. You can start ebusd with these parameters --checkconfig --dumpconfig=csv –dumpconfigto=/data/dump2.csv. The result was a good source for me, maybe for you too

**Reply by fabian1512** [reply]
great, thank you - found the wiki as well. you can loose a lot of time with that try  and error
